pkg/cmd/cli/cmd/expose.go

```diff
@@ -46,6 +51,14 @@ func validate(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
 	if cmdutil.GetFlagString(cmd, "generator") != "route/v1" {
 		return nil
 	}
+	// We don't want to use any other name for our route other than
+	// the one the service we expose has
+	providedName := cmdutil.GetFlagString(cmd, "name")
```

Comment:
Currently there is no other way to pass parameters into the route generation other than a [map](https://github.com/openshift/origin/blob/f11daef358df0f3424acb9e3cf39cafacfbff369/pkg/route/generator/generate.go#L16). These parameters are being set up by expose's flags and there is only one *name* parameter we can pass.:)